libs/async/include/hpx/async.hpp

```diff
@@ -1,15 +0,0 @@
-//  Copyright (c) 2020 ETH Zurich
-//
-//  SPDX-License-Identifier: BSL-1.0
-//  Distributed under the Boost Software License, Version 1.0. (See accompanying
-//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
-
-#pragma once
-
-#include <hpx/async/apply.hpp>
-#include <hpx/async/async.hpp>
-#include <hpx/async/async_callback.hpp>
-#include <hpx/async/async_continue.hpp>
-#include <hpx/async/async_continue_callback.hpp>
-#include <hpx/async/dataflow.hpp>
-#include <hpx/async/sync.hpp>
```

Comment:
This one we should keep. The `hpx/include` headers are one of the few things that we can 100% guarantee to be the same as in 1.4.1.

Reply:
I thought the one we keep is `libs/include/include/hpx/include/async.hpp` which I did not rename, does the one is `async_distributed` (`libs/async_distributed/include/hpx/async.hpp`) also is garanteed ?

Reply:
Yes, thanks for correcting me! I mentally skipped `async` in `libs/async/include` and thought this was in the `include` module. Is `hpx/async_distributed.hpp` generated now?

Reply:
I did not auto-generate it for now as there is 16 headers in the `async_distributed` module (excluding the detail and fwd ones) and only 7 in `hpx/async_distributed.hpp`, should I generate it ?

Reply:
Nope, let's keep it like this (again, I missed that this was moved, and not just deleted).